superset/common/query_object.py

```diff
@@ -83,10 +83,7 @@ def __init__(
 
         # Temporal solution for backward compatability issue
         # due the new format of non-ad-hoc metric.
-        self.metrics = [
-            metric if "expressionType" in metric else metric["label"]  # type: ignore
-            for metric in metrics
-        ]
+        self.metrics = [utils.get_metric_name(metric) for metric in metrics]
```

Comment:
During testing I noticed that the existing logic was incomplete; `utils.get_metric_name` on the other hand is used elsewhere and handles all metric types correctly (legacy and ad-hoc).